src/commands/LoadController.php

```diff
@@ -184,12 +184,35 @@ protected function process($data)
             }
         }
 
-        foreach (ArrayHelper::getValue($data, 'assign', []) as $userid => $permissionOrRoles) {
+        $assignData = ArrayHelper::getValue($data, 'assign', []);
+        $findBy = ArrayHelper::remove($assignData, 'findBy');
+        foreach ($assignData as $userid => $permissionOrRoles) {
+            Yii::info(sprintf("User %s start:", $userid));
+            if(!is_null($findBy)) {
+                try {
+                    $userInfo = call_user_func_array([ Yii::$app->user->identityClass, $findBy ], [ $userid ]);
+                } catch (\ErrorException $exception) { // Catch function doesn't exist
+                    Yii::error(sprintf("%s", $exception->getMessage()));
+                    exit();
+                } catch (yii\authclient\InvalidResponseException $exception) { // Catch query response code 404
+                    Yii::warning(sprintf("%s", $exception->getMessage()));
+                    continue;
+                }
```

Comment:
We don't need the code before anymore we just need to check is user identity is null like done under.

Reply:
If we don't have this code, we cannot skip current user and working next because the program is stopped after throws error.

Reply:
Just FYI, the code from line 200 to 203 is preventing the case of `findBy: 'findByUsername'`. Because it does not trigger any exception, but return the data like 
`[
    'id' => null,
    'username' => null,
    'firstName' => null,
    'lastName' => null,
    'email' => null,
    'timezone' => 'UTC',
    'language' => 'en-US',
]`. So I suggest we need to improve the function `findByUsername()` in the keycloak module to make it same return behavior as function `findIdentity()` by return exception instead of variable with empty parameters.

Reply:
You are right but it's not related to that extension it's independent, the identity interface is saying that null is return if the user doesn't exist we need to stick on it. And if the findByUsername doesn't behave properly it needs in fact to be fixed.